infra/lib/logging/stack.ts

```diff
@@ -0,0 +1,162 @@
+    constructor(
+        scope: Construct,
+        environments: LoggingIntegrationTestStackEnvironmentProps[],
+        props?: cdk.NestedStackProps,
+    ) {
+        super({
+            scope,
+            category: AmplifyCategory.Logging,
+            environments,
+            props,
+        })
+    }
+
+    protected buildEnvironments(
+        environments: LoggingIntegrationTestStackEnvironmentProps[],
+    ): LoggingIntegrationTestStackEnvironment[] {
+        return environments.map((envProps) => {
+            return new LoggingIntegrationTestStackEnvironment(
+                this,
+                this.baseName,
+                envProps,
+            )
+        })
+    }
+}
+export class LoggingIntegrationTestStackEnvironment extends IntegrationTestStackEnvironment<LoggingIntegrationTestStackEnvironmentProps> {
+    constructor(
+        scope: Construct,
+        baseName: string,
+        props: LoggingIntegrationTestStackEnvironmentProps,
+    ) {
+        super(scope, baseName, props);
+
+        const region = cdk.Stack.of(this).region
+        const account = cdk.Stack.of(this).account
+
+        const logGroup = new logs.LogGroup(this, 'Log Group', {
+                retention: logs.RetentionDays.INFINITE
+            })
+
+        const identityPool = new cognito_identity.IdentityPool(this, "IdentityPool", {
+                allowUnauthenticatedIdentities: true,
+            });
+        
+        const authRole = identityPool.authenticatedRole;
+        const unAuthRole = identityPool.unauthenticatedRole;
+
+        const logResource = `arn:aws:logs:${region}:${account}:log-group:${logGroup.logGroupName}:log-stream:*`
+        const logIAMPolicy = new iam.PolicyStatement({
+            effect: iam.Effect.ALLOW,
+            resources: [logResource],
+            actions: ["logs:PutLogEvents", "logs:DescribeLogStreams", "logs:CreateLogStream"]
+        })
+
+        authRole.addToPrincipalPolicy(logIAMPolicy)
+        unAuthRole.addToPrincipalPolicy(logIAMPolicy)
+
+        // remote config
+        if (props.remoteConfig) {
+
+            const loggingConfigLocation = 'lib/logging/resources/config/remoteloggingconstraints.json';
+            const lambdaConfig = 'lib/logging/resources/lambda/remoteconfig.js';
+            const configFileName = 'lib/loggingremoteconfig.json';
+
+            // Create a bucket for the remote config
+            const remoteConfigBucket = new S3.Bucket(this, 'AmplifyRemoteLogging-Bucket', {
+                publicReadAccess: false,
+                versioned: true,
+                removalPolicy: cdk.RemovalPolicy.DESTROY,
+                enforceSSL: true,
+                blockPublicAccess: S3.BlockPublicAccess.BLOCK_ALL,
+            });
+
+            // Deploy the remote config to the bucket
+            new BucketDeployment(this, 'AmplifyRemoteLogging-BucketDeployment', {
+                sources: [
+                    Source.asset(path.dirname(path.join(loggingConfigLocation))),
```

Comment:
This path logic is confusing. Is this saying upload the directory or the file? `join` is not needed here. And `dirname` makes specifying the full path unnecessary.

Reply:
If I remove the `path.dirname(path.join(...)` completely, I get an error saying `Asset path must be either a .zip file or a directory`. I'm able to run this with just `Source.asset(path.dirname(loggingConfigLocation)` or I could run it without `path.dirname` if I change the `loggingConfigLocation` to `lib/logging/resources/config/` from `lib/logging/resources/config/remoteloggingconstraints.json`